util/bitstream_test.go

```diff
@@ -16,6 +16,28 @@ type testDefinition struct {
 	value  uint64 // The value we expect the function to return (64 bit to allow for future functions that extract larger ints)
 }
 
+var test1Bit = []testDefinition{
+	{testdata, 2, 0},  // testdata 0 in first byte
+	{testdata, 5, 1},  // testdata 1 in first byte
+	{testdata, 23, 1}, // testdata 1 in last bit of third byte
+	{testdata, 47, 1}, // testdata 1 in last bit of last byte
+}
+
+func TestReadByte1(t *testing.T) {
+	bs := BitStream{b: testdata, p: 80}
+
+	b, err := bs.ReadByte1()
+	assertStringsEqual(t, "Expected 1 bit at bit 80, but the byte array was only 6 bytes long", err.Error())
+
+	for _, test := range test1Bit {
+		bs = BitStream{b: test.data, p: test.offset}
```

Comment:
I discovered a testing pattern we haven't used before. We might be better off using t.Run(description, anon func) to isolate each test in the loop. There's some cool tooling around the use of t.Run as well which might be useful to us in the future.